src/mpid/ch4/src/ch4_request.h

```diff
@@ -89,7 +89,7 @@ MPL_STATIC_INLINE_PREFIX int MPID_Request_complete(MPIR_Request * req)
         if (req->completion_notification)
             MPIR_cc_dec(req->completion_notification);
 
-        if (MPIDIG_REQUEST(req, req)) {
```

Comment:
Why checking whether `MPIDIG_REQUEST(req, req)` is `NULL` isn't sufficient?

Reply:
`MPIDIG_REQUEST(req, req)` is `NULL `is a code smell.

Reply:
I mean checking whether the value is `NULL ` for other purposes.